### PR TITLE
feat/pdk.kong.request.cookie

### PR DESCRIPTION
**Pull request by fairyqb**
Summary

add kong.request.cookie(name) and cookies()

### CODE REVIEW

**Comment by fairyqb** [comment]
request to merge ? @Tieske  @bungle

**Comment by Tieske** [comment]
@fairyqb I did leave some comments, which haven't been resolved yet.

File: kong/pdk/request.lua [hunk 1]
```diff
@@ -603,20 +616,29 @@ local function new(self)
...
     return ngx.req.get_headers(max_headers)
   end
-  
+
 
   ---
-  -- Returns a Lua table holding the request cookie.
+  ---
+  -- Returns a Lua table holding the request cookies. Keys are cookie names.
+  -- Values are either a string with the cookie value.
```
Comment (Tieske) [inline kong/pdk/request.lua:624]:
```suggestion
  -- Values are strings with the cookie value.
```

**Review by Tieske — CHANGES_REQUESTED** [review]
minor typo

**Comment by fairyqb** [comment]
@Tieske , finished, thank you.

File: kong/pdk/request.lua [hunk 2]
```diff
@@ -491,6 +492,38 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the value of the specified request cookie.
+  --
+  -- The returned value is either a `string`, or can be `nil` if a cookie with
+  -- `name` was not found in the request.
+  --
+  -- Cookie names in are case-sensitive.
```
Comment (Tieske) [inline kong/pdk/request.lua:501]:
```suggestion
  -- Cookie names are case-sensitive.
```

File: kong/pdk/request.lua [hunk 3]
```diff
@@ -585,6 +618,31 @@ local function new(self)
   end
 
 
+  ---
+  ---
```
Comment (Tieske) [inline kong/pdk/request.lua:622]: two lines with `---`

**Review by Tieske — CHANGES_REQUESTED** [review]
sorry, 2 more minor typo things

**Comment by Tieske** [comment]
@bungle would you mind also having a look at this one? (also; shouldn't this be targetted on `next` instead of `master`)

File: kong/pdk/request.lua (hunk 3 above)
Comment (fairyqb) [reply kong/pdk/request.lua]: ok.

File: kong/pdk/request.lua (hunk 2 above)
Comment (fairyqb) [reply kong/pdk/request.lua]: thank you.

**Comment by fairyqb** [comment]
> @bungle would you mind also having a look at this one? (also; shouldn't this be targetted on `next` instead of `master`)

#6273

**Comment by Tieske** [comment]
@fairyqb no need to send a second pr, you can rebase your branch on the next branch and force push, then in the github ui edit the pr (button at top right) and change the target branch

**Comment by fairyqb** [comment]
I got it

**Pull request closed without merging**